### PR TITLE
Initial removal of support for Ohai V6 plugins

### PR DESCRIPTION
**Pull request by tas50**
This doesn't refactor anything. Just removes the old support in the most brutal way possible. With this in we can start the work to refactor the loader and runner code.

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

**Comment by tas50** [comment]
I think this may be a MVP of ripping out V6 w/o refactoring any code. I'd appreciate any feedback.

File: lib/ohai/loader.rb [hunk 1]
```diff
@@ -133,11 +124,8 @@ def load_plugin_class(plugin_path, plugin_dir_path = nil)
         load_v7_plugin_class(contents, plugin_path)
       else
         Ohai::Log.warn("[DEPRECATION] Plugin at #{plugin_path} is a version 6 plugin. \
```
Comment (thommay) [inline lib/ohai/loader.rb]: I don't think this is a warn any more, this is a fatal error.

File: lib/ohai/runner.rb [hunk 2]
```diff
@@ -45,8 +45,6 @@ def run_plugin(plugin)
           case plugin.version
           when :version7
             run_v7_plugin(plugin)
-          when :version6
-            run_v6_plugin(plugin)
```
Comment (thommay) [inline lib/ohai/runner.rb]: we'd be better off just raising unless this is a v7 plugin, and ditching this whole case statement i think.

File: lib/ohai/system.rb [hunk 3]
```diff
@@ -87,19 +85,6 @@ def load_plugins
     end
 
     def run_plugins(safe = false, attribute_filter = nil)
```
Comment (thommay) [inline lib/ohai/system.rb]: this whole function can get cleaned up a lot.

Comment (tas50) [reply lib/ohai/system.rb]: There's several methods where that's true

**Pull request merged**